manager/controlapi/node_test.go

```diff
@@ -523,45 +523,65 @@ func TestUpdateNodeDemote(t *testing.T) {
 		return nil
 	}))
 
-	// Try to demote Node 2
-	r, err = ts.Client.GetNode(context.Background(), &api.GetNodeRequest{NodeID: nodes[2].SecurityConfig.ClientTLSCreds.NodeID()})
+	var demoteNode, lastNode *raftutils.TestNode
+	if leader {
+		demoteNode = nodes[1]
+		lastNode = nodes[2]
+	} else {
+		demoteNode = nodes[2]
+		lastNode = nodes[1]
+	}
+
+	// Try to demote a Node and scale down to 1
+	r, err = ts.Client.GetNode(context.Background(), &api.GetNodeRequest{NodeID: demoteNode.SecurityConfig.ClientTLSCreds.NodeID()})
 	assert.NoError(t, err)
 	spec = r.Node.Spec.Copy()
 	spec.Role = api.NodeRoleWorker
 	version = &r.Node.Meta.Version
 	_, err = ts.Client.UpdateNode(context.Background(), &api.UpdateNodeRequest{
-		NodeID:      nodes[2].SecurityConfig.ClientTLSCreds.NodeID(),
+		NodeID:      demoteNode.SecurityConfig.ClientTLSCreds.NodeID(),
 		Spec:        spec,
 		NodeVersion: version,
 	})
 	assert.NoError(t, err)
 
+	// Update the server
+	ts.Server.raft = lastNode.Node
+	ts.Server.store = lastNode.MemoryStore()
+
 	newCluster = map[uint64]*raftutils.TestNode{
-		1: nodes[1],
+		1: lastNode,
 	}
 
 	raftutils.WaitForCluster(t, clockSource, newCluster)
 
-	// New server should list 1 member
 	assert.NoError(t, raftutils.PollFunc(clockSource, func() error {
-		members := nodes[1].GetMemberlist()
+		members := lastNode.GetMemberlist()
 		if len(members) != 1 {
 			return fmt.Errorf("expected 1 node, got %d", len(members))
 		}
 		return nil
 	}))
 
 	// Make sure we can't demote the last manager.
-	r, err = ts.Client.GetNode(context.Background(), &api.GetNodeRequest{NodeID: nodes[1].SecurityConfig.ClientTLSCreds.NodeID()})
+	r, err = ts.Client.GetNode(context.Background(), &api.GetNodeRequest{NodeID: lastNode.SecurityConfig.ClientTLSCreds.NodeID()})
 	assert.NoError(t, err)
 	spec = r.Node.Spec.Copy()
 	spec.Role = api.NodeRoleWorker
 	version = &r.Node.Meta.Version
 	_, err = ts.Client.UpdateNode(context.Background(), &api.UpdateNodeRequest{
-		NodeID:      nodes[1].SecurityConfig.ClientTLSCreds.NodeID(),
+		NodeID:      lastNode.SecurityConfig.ClientTLSCreds.NodeID(),
 		Spec:        spec,
 		NodeVersion: version,
 	})
 	assert.Error(t, err)
 	assert.Equal(t, codes.FailedPrecondition, grpc.Code(err))
```

Comment:
I'm not sure `controlapi` is the place that, but that seems like a good idea to include in the raft tests.